src/name_core.jl

```diff
@@ -127,6 +127,26 @@ function permute_dimnames(dimnames::NTuple{N, Symbol}, perm) where N
     return compile_time_return_hack(new_dimnames)
 end
 
+"""
+    _rename(dimnames::Tuple, old_new::Vararg{Pair})
+    _rename(dimname::Symbol, old_new::Vararg{Pair})
+
+For each pair `old=>new`, replace all occurences of `old` in tuple `namea` with `new`.
+If `dimname` is a Symbol, replace it with `new` of the pair with matching `old`, or
+return `dimname` if no pairs match.
+"""
+function _rename(dimnames::Tuple, old_new::Vararg{Pair})
+    return ntuple(i -> _rename(dimnames[i], old_new...), length(dimnames))
+end
+
+function _rename(dimname::Symbol, old_new::Vararg{Pair})
+    # Avoid looping over pairs explicitly because that allocates.
+    nt = ntuple(i -> dimname === first(old_new[i]) ? i : 0, length(old_new))
+    which = sum(nt)
+    which > length(old_new) && throw(ArgumentError("`old_new` needs unique `old` items"))
```

Comment:
Can this error message be made clearer?
Maybe `duplicate old names are not permitted in rename` or something?>

Reply:
thanks, that's much better